src/psyclone/tests/domain/lfric/arg_index_to_metadata_index_test.py

```diff
@@ -0,0 +1,175 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: R. W. Ford, STFC Daresbury Lab
+
+'''This module implements tests for the ArgIndexToMetadataIndex
+class.
+
+'''
+from psyclone.domain.lfric import ArgIndexToMetadataIndex
+from psyclone.domain.lfric.kernel import (
+    ScalarArgMetadata, FieldArgMetadata, FieldVectorArgMetadata,
+    OperatorArgMetadata, ColumnwiseOperatorArgMetadata, LFRicKernelMetadata)
+
+# pylint: disable=unsubscriptable-object
+
+
+def call_method(method_name, meta_arg, metadata):
+    '''Utility method to initialise the required class variables and then
+    call the method specified in the method_name argument.
+
+    :param str method_name: the name of the method to call.
+    :param meta_arg: the meta_arg metadata to pass to the call.
+    :type meta_arg: subclass of \
+        :py:class:`psyclone.domain.lfric.kernel.CommonArgMetadata`
+    :param metadata: the full metadata description.
+    :type metadata: \
+        :py:class:`psyclone.domain.lfric.kernel.LFRicKernelMetadata`
+
+    :returns: an ArgIndexToMetadataIndex class
+    :rtype: :py:class:`psyclone.domain.lfric.ArgIndexToMetadataIndex`
+
+    '''
+    metadata.validate()
+    cls = ArgIndexToMetadataIndex
+    cls._initialise(None)
+    cls._metadata = metadata
+    getattr(cls, method_name)(meta_arg)
+    return cls
+
+
+def test_initialise():
+    '''Test the _initialise method.'''
+    cls = ArgIndexToMetadataIndex
+    cls._index = 2
+    cls._info = None
+    cls._initialise(None)
+    assert cls._index == 0
+    assert isinstance(cls._info, dict)
```

Comment:
This has really got me stumped. Where does cls._info get set to be a `dict`?

Reply:
OK, I finally tracked it down. Please see comments in implementation file.

Reply:
By "implementation file" I meant the source file that this test file is for. Not that it matters now...